File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F05.P2.cpp
#include <cstdint>
#include <cstdio>

std::int32_t probe() {
// BEGIN PROBE F05.P2
auto put = [](std::int32_t& out) { out = 12; };
std::int32_t cell = 3;
put(cell);
return cell;
// END PROBE F05.P2
}

int main() {
    std::printf("%d\n", probe());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F05.P2.go
package main

import "fmt"

func probe() int32 {
	// BEGIN PROBE F05.P2
	put := func(p *int32) { *p = 12 }
	var cell int32 = 3
	put(&cell)
	return cell
	// END PROBE F05.P2
}

func main() {
	fmt.Println(probe())
}
